src/shared/ui/Counter/Counter.test.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, screen } from "@testing-library/react";
+import userEvent from '@testing-library/user-event'
+import { act } from 'react-dom/test-utils'
+import { Counter } from "./Counter";
+
+describe("Counter Component Functionality", () => {
+    test("Should render initial value", () => {
+        const initialValue = 5;
+        render(<Counter initialValue={initialValue} />);
+        const initialValueElement = screen.getByText(initialValue);
+        expect(initialValueElement).toBeInTheDocument();
```

Comment:
the same result can be achieved with a shorter code
```suggestion
        screen.getByText(initialValueElement)
```

Reply:
it didn`t work in my case(

Reply:
take a look at my commit, I fixed it